pkg/operator/ceph/csi/spec.go

```diff
@@ -453,14 +484,104 @@ func (r *ReconcileCSI) startDrivers(ver *version.Info, ownerInfo *k8sutil.OwnerI
 			return errors.Wrap(err, "failed to load nfs provisioner deployment template")
 		}
 	}
+	multusApplied := len(r.multusClusters) > 0
 
 	// get common provisioner tolerations and node affinity
 	provisionerTolerations := getToleration(r.opConfig.Parameters, provisionerTolerationsEnv, []corev1.Toleration{})
 	provisionerNodeAffinity := getNodeAffinity(r.opConfig.Parameters, provisionerNodeAffinityEnv, &corev1.NodeAffinity{})
+
 	// get common plugin tolerations and node affinity
 	pluginTolerations := getToleration(r.opConfig.Parameters, pluginTolerationsEnv, []corev1.Toleration{})
 	pluginNodeAffinity := getNodeAffinity(r.opConfig.Parameters, pluginNodeAffinityEnv, &corev1.NodeAffinity{})
 
+	if multusApplied && !v.SupportsMultus() {
+		// TODO: uncomment the return line and remove the logging line
+		logger.Errorf("multus is applied but the csi version %q does not support it, need at least %q", v.String(), multusSupportedVersion.String())
+		//return errors.Errorf("multus is applied but the csi version %q does not support it, need at least %q", v.String(), multusSupportedVersion.String())
+	}
+
+	for _, multusCluster := range r.multusClusters {
+		for _, driver := range enabledDrivers {
+			cephPluginHolder, err := templateToDaemonSet("cephpluginholder", driver.holderTemplate, tp)
+			if err != nil {
+				return errors.Wrapf(err, "failed to load ceph %q plugin holder template", driver.fullName)
+			}
+
+			// DO NOT set owner reference on ceph plugin holder daemonset, this DS must never restart unless
+			// the entire node is rebooted
+			holderPluginTolerations := getToleration(r.opConfig.Parameters, driver.toleration, pluginTolerations)
+			holderPluginNodeAffinity := getNodeAffinity(r.opConfig.Parameters, driver.nodeAffinity, pluginNodeAffinity)
+
+			// apply driver's plugin tolerations and node affinity
+			applyToPodSpec(&cephPluginHolder.Spec.Template.Spec, holderPluginNodeAffinity, holderPluginTolerations)
+
+			// apply resource request and limit to *** plugin containers
+			// TODO: expose resources or inherit? the holder needs lesser resources...
+			applyResourcesToContainers(r.opConfig.Parameters, cephFSPluginResource, &cephPluginHolder.Spec.Template.Spec)
+
+			// Append the CEPH_FSID env var so that the main container can use it to create the network
+			// namespace symlink to the Kubelet plugin directory
+			cephPluginHolder.Spec.Template.Spec.Containers[0].Env = append(
+				cephPluginHolder.Spec.Template.Spec.Containers[0].Env,
+				corev1.EnvVar{
+					Name:  "CEPH_FSID",
```

Comment:
This will also work, instead of CEPH_FSID use the ceph cluster namespace name? the cluster namespace name is the unique identifier (clusterID) in the configmap.

Reply:
That's also easier than passing clusterInfo around too. I'll change that, thanks!

Reply:
@Madhu-1 Did you mean just the clusterName instead of cluster namespace name?

Reply:
I meant cluster namespace name. if we have multiple clusters,  the cluster name can be the same as it's a namespace resource right?

Reply:
Today when I look at the clusterID, this is only either the name of the namespace not the namespace name

Reply:
Sorry for the confusion, I meant namespace name where ceph cluster CR is created.